### PR TITLE
Merge 3.4.x into 4.0.x

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: src/Schema/Comparator.php [hunk 1]
```diff
@@ -21,6 +21,8 @@
 
 /**
  * Compares two Schemas and return an instance of SchemaDiff.
+ *
+ * @method SchemaDiff compareSchemas(Schema $fromSchema, Schema $toSchema)
```
Comment (derrabus) [inline src/Schema/Comparator.php]: This feels redundant now.

File: tests/Schema/ComparatorTest.php [hunk 2]
```diff
@@ -16,13 +16,15 @@
...
 use Doctrine\DBAL\Types\Type;
+use Doctrine\Deprecations\PHPUnit\VerifyDeprecations;
 use PHPUnit\Framework\TestCase;
 
 use function array_keys;
 
 abstract class ComparatorTest extends TestCase
 {
     protected Comparator $comparator;
+    use VerifyDeprecations;
```
Comment (derrabus) [inline tests/Schema/ComparatorTest.php]: The trait is not used anymore, is it?

File: src/Schema/Comparator.php [hunk 3]
```diff
@@ -39,8 +39,10 @@ public function __construct(AbstractPlatform $platform)
      *
      * @throws SchemaException
      */
-    public function compareSchemas(Schema $fromSchema, Schema $toSchema): SchemaDiff
-    {
+    public function compareSchemas(
+        Schema $fromSchema,
+        Schema $toSchema
+    ): SchemaDiff {
```
Comment (morozov) [inline src/Schema/Comparator.php]: This could be reverted.

File: phpstan.neon.dist [hunk 4]
```diff
@@ -97,6 +97,10 @@ parameters:
             message: '~^Parameter #2 \$lockMode of method Doctrine\\DBAL\\Platforms\\AbstractPlatform\:\:appendLockHint\(\) expects 0\|1\|2\|4, 128 given\.$~'
             path: tests/Platforms/AbstractPlatformTestCase.php
 
+        # Type check for legacy implementations of the Connection interface
+        # TODO: remove in 4.0.0
+        - "~Call to function method_exists\\(\\) with Doctrine\\\\DBAL\\\\Driver\\\\Connection and 'getNativeConnection' will always evaluate to true\\.~"
```
Comment (morozov) [inline phpstan.neon.dist]: This patch adds an error suppression but doesn't seem to be adding the code that causes the error. If this line is removed, the build still passes.

**Review by morozov — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Pull request merged**